core-api/src/main/kotlin/com/jakemoore/datakache/core/connections/mongo/changestream/ChangeStreamEventProcessor.kt

```
@@ -0,0 +1,386 @@
+    fun resetCountersForRestart() {
+        // Reset events counter to prevent overflow and provide fresh start
+        totalEventsProcessed = 0L
+        lastTokenCleanupTime = System.currentTimeMillis()
+        context.logger.debug("Reset counters for restart")
+    }
+
+    /**
+     * Starts the event processor that handles events from the channel with timeout and backpressure.
+     */
+    @OptIn(ExperimentalCoroutinesApi::class, DelicateCoroutinesApi::class)
+    fun startEventProcessing(scope: kotlinx.coroutines.CoroutineScope): Job {
```

detekt (1.23.8)

[warning] The function startEventProcessing appears to be too complex based on Cognitive Complexity (complexity: 30). Defined complexity threshold for methods is set to '25'

(detekt.complexity.CognitiveComplexMethod)